### PR TITLE
@liveblocks/react-comments initial docs

### PR DESCRIPTION
**Pull request by GuillaumeSalles**

### CODE REVIEW

File: docs/pages/api-reference/liveblocks-react-comments.mdx [hunk 1]
```diff
@@ -0,0 +1,852 @@
...
+IDs matching that text. This function will be called every time the text changes
+but with some debouncing.
+
+```tsx file="liveblocks.config.ts"
+// ...
+
+export const {
+  /* ... */
+} = createRoomContext(client, {
+  resolveMentionSuggestions: (text) => __getSuggestedUserIds__(text),
```
Comment (CTNicholas) [inline docs/pages/api-reference/liveblocks-react-comments.mdx:118]: Similar to above, can we show what we're returning (then perhaps have a collapsed more realistic example below).

File: docs/pages/api-reference/liveblocks-react-comments.mdx [hunk 2]
```diff
@@ -0,0 +1,852 @@
...
+
+Returns a function that creates a thread with an initial comment, and optionally
+some metadata.
+
+```tsx
+const createThread = useCreateThread();
+const thread = createThread({ body: {}, metadata: {} });
+```
+
+### useEditThread
```
Comment (CTNicholas) [inline docs/pages/api-reference/liveblocks-react-comments.mdx:210]: Can threads be deleted too? No hook for this is mentioned. What if every comment in a thread is deleted?

**Review by CTNicholas — COMMENTED** [review]
Excellent work on this, this is great for a private beta! Just recommending a few small changes. When we get close to actual release, I'll spend lots of time carefully reviewing wording, and everything else.

File: docs/pages/api-reference/liveblocks-react-comments.mdx (hunk 2 above)
Comment (marcbouchenoire) [reply docs/pages/api-reference/liveblocks-react-comments.mdx:210]: They can't be manually deleted no, only comments can. If every comment in a thread is deleted, the thread is automatically deleted.

Comment (CTNicholas) [reply docs/pages/api-reference/liveblocks-react-comments.mdx:214]: > They can't be manually deleted no, only comments can. If every comment in a thread is deleted, the thread is automatically deleted.

@marcbouchenoire Right! Could you make a note of that under `ThreadData`?

File: docs/pages/api-reference/liveblocks-react-comments.mdx [hunk 3]
```diff
@@ -0,0 +1,852 @@
...
+```tsx file="liveblocks.config.ts"
+// ...
+
+export const {
+  /* ... */
+} = createRoomContext(client, {
+  resolveUser: (userId) => __getUserInfoFromDB__(userId),
+});
+```
+
```
[the remark this answers is not in the document]
Comment (marcbouchenoire) [reply docs/pages/api-reference/liveblocks-react-comments.mdx:105]: I've tweaked the snippet and added a note below to explain in more details what's expected.

File: docs/pages/api-reference/liveblocks-react-comments.mdx (hunk 1 above)
Comment (marcbouchenoire) [reply docs/pages/api-reference/liveblocks-react-comments.mdx]: I've tweaked the snippet.

File: docs/pages/api-reference/liveblocks-react-comments.mdx [hunk 4]
```diff
@@ -0,0 +1,852 @@
...
+  createdAt: string;
+  updatedAt?: string;
+  comments: CommentData[];
+  metadata: ThreadMetadata;
+};
+```
+
+### CommentData
+
+Represents a comment.
```
[the remark this answers is not in the document]
Comment (GuillaumeSalles) [reply docs/pages/api-reference/liveblocks-react-comments.mdx:189]: 👍

File: docs/pages/api-reference/liveblocks-react-comments.mdx (hunk 2 above)
Comment (GuillaumeSalles) [reply docs/pages/api-reference/liveblocks-react-comments.mdx:248]: I added a small note under `ThreadData`.

Happy to iterate more tomorrow.

**Review by FlowFlorent — APPROVED** [review]

**Pull request merged**